python_modules/dagster/dagster/core/definitions/solid_definition.py

```diff
@@ -323,6 +331,7 @@ def get_resource_requirements(
                         key=io_manager_key,
                         node_description=node_description,
                         input_name=input_def.name,
+                        root_input=False,  # TODO maybe need to be true
```

Comment:
if we keep the `root_input` attribute in InputManagerRequirement, need to make sure that this is set correctly for the asset case

Reply:
@dpeng817 I want to make sure setting `root_input=False` makes sense wrt the condition here https://github.com/dagster-io/dagster/pull/8170/files#r891736416